From 6573bc729bd4e10116fc789a525f4133376b42c4 Mon Sep 17 00:00:00 2001
From: Ben Hilburn <ben.hilburn@ettus.com>
Date: Fri, 28 Aug 2015 12:07:00 -0700
Subject: [PATCH 07/17] Fixes Coverity Defect 1229872: Memory leak in
 `atsci_equalizer_lms2` DTOR

Simple issue of a missing `fclose` on an open file handle causing a memory leak.
---
 gr-atsc/lib/atsci_equalizer_lms2.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gr-atsc/lib/atsci_equalizer_lms2.cc b/gr-atsc/lib/atsci_equalizer_lms2.cc
index 5109793..73d93e6 100644
--- a/gr-atsc/lib/atsci_equalizer_lms2.cc
+++ b/gr-atsc/lib/atsci_equalizer_lms2.cc
@@ -90,6 +90,7 @@ atsci_equalizer_lms2::atsci_equalizer_lms2 ()
 
 atsci_equalizer_lms2::~atsci_equalizer_lms2 ()
 {
+  fclose(trainingfile);
 }
 
 void
-- 
2.1.4

